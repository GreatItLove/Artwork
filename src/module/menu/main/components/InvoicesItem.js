import React from 'react';
import { ListItem } from 'material-ui/List';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

import { INVOICES_PATH } from '../../../../config/routesPatch';

const propTypes = {
  style: PropTypes.object.isRequired
};

/**
 * @function InvoicesItem
 * @param {Object} style
 * @returns {XML}
 */
function InvoicesItem({ style }) {
  return (
    <ListItem
      key={INVOICES_PATH}
      primaryText="Invoices"
      innerDivStyle={style}
      leftIcon={<i className="material-icons">payment</i>}
      containerElement={<NavLink to={INVOICES_PATH} exact />}
    />
  );
}

InvoicesItem.propTypes = propTypes;

export default InvoicesItem;
