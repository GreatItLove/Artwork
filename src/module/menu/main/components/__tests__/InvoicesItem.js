import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import InvoicesItem from '../InvoicesItem';

describe('InvoicesItem', () => {
  it('Snapshot of InvoicesItem', () => {
    const wrapper = shallow(<InvoicesItem style={{}} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
