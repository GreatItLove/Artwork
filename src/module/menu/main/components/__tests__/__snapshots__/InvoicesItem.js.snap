// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoicesItem Snapshot of InvoicesItem 1`] = `
<ListItem
  autoGenerateNestedIndicator={true}
  containerElement={
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      exact={true}
      to="/invoices"
    />
  }
  disableKeyboardFocus={false}
  disabled={false}
  initiallyOpen={false}
  innerDivStyle={Object {}}
  insetChildren={false}
  leftIcon={
    <i
      className="material-icons"
    >
      payment
    </i>
  }
  nestedItems={Array []}
  nestedLevel={0}
  onKeyboardFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onNestedListToggle={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  open={null}
  primaryText="Invoices"
  primaryTogglesNestedList={false}
  secondaryTextLines={1}
/>
`;
